intelmq/bin/intelmqctl.py

```diff
@@ -1448,32 +1441,21 @@ def check(self, no_connections=False):
                 message = "Bot %r has invalid `run_mode` %r. Must be 'continuous' or 'scheduled'."
                 check_logger.warning(message, bot_id, bot_config['run_mode'])
                 retval = 1
-            if bot_id not in files[PIPELINE_CONF_FILE] and bot_config.get('enabled', True):
-                check_logger.error('Misconfiguration: No pipeline configuration found for %r.', bot_id)
-                retval = 1
-            elif bot_id not in files[PIPELINE_CONF_FILE] and not bot_config.get('enabled', True):
-                check_logger.warning('Misconfiguration: No pipeline configuration found for %r.', bot_id)
-            elif bot_id in files[PIPELINE_CONF_FILE]:
-                if ('group' in bot_config and
-                        bot_config['group'] in ['Collector', 'Parser', 'Expert']):
-                    if ('destination-queues' not in files[PIPELINE_CONF_FILE][bot_id] or
-                            (isinstance(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'], list) and
-                             len(files[PIPELINE_CONF_FILE][bot_id]['destination-queues']) < 1) or
-                            (isinstance(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'], dict) and
-                             '_default' not in files[PIPELINE_CONF_FILE][bot_id]['destination-queues'])):
-                        check_logger.error('Misconfiguration: No (default) destination queue for %r.', bot_id)
-                        retval = 1
-                    else:
-                        all_queues = all_queues.union(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'])
-                if ('group' in bot_config and
-                        bot_config['group'] in ['Parser', 'Expert', 'Output']):
-                    if ('source-queue' not in files[PIPELINE_CONF_FILE][bot_id] or
-                            not isinstance(files[PIPELINE_CONF_FILE][bot_id]['source-queue'], str)):
-                        check_logger.error('Misconfiguration: No source queue for %r.', bot_id)
-                        retval = 1
-                    else:
-                        all_queues.add(files[PIPELINE_CONF_FILE][bot_id]['source-queue'])
-                        all_queues.add(files[PIPELINE_CONF_FILE][bot_id]['source-queue'] + '-internal')
+            if ('group' in bot_config and bot_config['group'] in ['Collector', 'Parser', 'Expert']):
+                if ('destination-queues' not in bot_config['parameters'] or
+                   (isinstance(bot_config['parameters']['destination_queues'], list) and len(bot_config['parameters']['destination-queues']) < 1) or
```

Comment:
If I understand the code in `bot.py` correctly, the list type is no longer supported, only the dict type?

Reply:
I decided to make the patch as small as possible. The deprecation of the list and string type would also lead to changes in  the `pipeline.py` and I think its better to implement that as a separate PR.